src/server/pfs/db/driver.go

```diff
@@ -125,13 +126,34 @@ func initDB(session *gorethink.Session, dbName string) error {
 		return nil
 	}
 
-	// Create tables
-	for _, table := range tables {
-		tableCreateOpts := tableToTableCreateOpts[table]
-		if _, err := gorethink.DB(dbName).TableCreate(table, tableCreateOpts...).RunWrite(session); err != nil {
-			return err
+	// There is a race here
+	//
+	// When rethink is under load, we consistently fail on the table creation
+	// w a 'db not found' error right after we created it a few lines above.
+	//
+	// However, the db shows up fine in the dashboard, so it does in fact get
+	// created. This is only relevant for tests, because thats the only time
+	// we create a bunch of databases while R/W from rethink as well
+	config := backoff.NewExponentialBackOff()
+	// We want to backoff more aggressively (i.e. wait longer) than the default
+	config.InitialInterval = 1 * time.Second
+	config.Multiplier = 2
+	config.MaxElapsedTime = 5 * time.Minute
+	tablesToInitialize := make([]Table, len(tables))
+	copy(tablesToInitialize, tables)
+	backoff.RetryNotify(func() error {
+		// Create tables
+		for i, table := range tablesToInitialize {
+			tableCreateOpts := tableToTableCreateOpts[table]
+			if _, err := gorethink.DB(dbName).TableCreate(table, tableCreateOpts...).RunWrite(session); err != nil {
+				return err
+			}
+			tablesToInitialize = append(tablesToInitialize[:i], tablesToInitialize[i+1:]...)
```

Comment:
This line seems pretty dubious to me. I'm pretty sure this will encounter a bounds error won't it? This example hits a bounds error and seems pretty equivalent: https://play.golang.org/p/3xh6kad7An On the other hand CI is passing, which gives me pause since it suggests that maybe this code isn't being run? How big is `tables` here? The bounds error won't manifest is `len(tables) <= 1` but I'm pretty sure it's bigger than that isn't it? I could just be wrong about this being a bounds error... but like I said my example seems pretty identical.

The problem here is that you're trying to wrap the creation of __all__ the tables in a single retry loop which means you need to keep track of what you accomplished in previous attempts. Instead just wrap each `TableCreate` in its own retry loop. This should be the same amount of code just moved around a little and no need for list splicing.